main/mining_controller.c

```diff
@@ -0,0 +1,253 @@
+#include "mining_controller.h"
+#include "esp_log.h"
+#include "esp_system.h"
+#include "freertos/semphr.h"
+#include <stdlib.h>
+
+#include "asic_result_task.h"
+#include "asic_task.h"
+#include "create_jobs_task.h"
+#include "main.h"
+#include "power_management_task.h"
+#include "stratum_task.h"
+
+#include "asic.h"
+#include "serial.h"
+#include "vcore.h"
+#include "work_queue.h"
+
+#include "nvs_config.h"
+
+static const char * TAG = "mining_controller";
+
+esp_err_t start_mining(GlobalState * global_state)
```

Comment:
A general comment, it looks like this class has too much responsibility. Ideally, it should only toggle the flag, and start tasks, and the tasks themselves should handle all responsibilities. F.e. retrieving frequency and logging about it, that should be done inside POWER_MANAGEMENT_task.

Secondary remark: it looks like some stuff is duplicated from `main.c`. Ideally, the cold boot should use the exact same startup sequence and code as a restart.